grails-app/services/au/org/emii/gogoduck/job/JobExecutorService.groovy

```diff
@@ -1,11 +1,82 @@
 package au.org.emii.gogoduck.job
 
+import au.org.emii.gogoduck.worker.Worker
+
+import java.util.concurrent.LinkedBlockingQueue
+import java.util.concurrent.BlockingQueue
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
 class JobExecutorService {
 
+    static BlockingQueue<Map> JOB_QUEUE
```

Comment:
FYI, I did start out using an `ArrayBlockingQueue`, the main difference being that this implementation *must* have a size limit.  Although I think this is a good idea, it would've been a change of behaviour, so I chose this instead.  Let's discuss this further though, because as it stands, this queue could grow indefinitely and burn all the memory (although, TBH, I think that's unlikely unless someone intentionally tried to DoS us).